### PR TITLE
Activity hierarchy for business investment

### PR DESCRIPTION
**Pull request by MikeData**
### What

Added activity hierarchy.

### How to review

Load into Neo. Try loading the associated datasets.

### Who can review

Rob or Eleanor.

### CODE REVIEW

File: cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "H" and child.code = "H53" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "_T" and child.code = "I" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "_T" and child.code = "J" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "J" and child.code = "J58" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "J" and child.code = "J58T60" CREATE (child)-[:hasParent]->(parent);
```
Comment (eldeal) [inline cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher]: Should this be the one that links directly to J, and J58 and J59_60 go through this?

File: cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "K" and child.code = "K65" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "K" and child.code = "K66" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M_N" and child.code = "M" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M69T71" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M69_70" CREATE (child)-[:hasParent]->(parent);
```
Comment (eldeal) [inline cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher]: Should this for through the child above?

File: cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M71" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M72" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M73" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M73T75" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "M" and child.code = "M74_75" CREATE (child)-[:hasParent]->(parent);
```
Comment (eldeal) [inline cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher]: Should these go through the M73T75 as mentioned with others?

File: cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "N" and child.code = "N78" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "N" and child.code = "N79" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "N" and child.code = "N80T82" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "_T" and child.code = "O" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "O" and child.code = "OTQ" CREATE (child)-[:hasParent]->(parent);
```
Comment (eldeal) [inline cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher]: OTQ is a child of O? But not a child of P? Shouldnt O, P and Q each be a child of OTQ?

File: cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "Q" and child.code = "Q87_88" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "_T" and child.code = "R" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "R" and child.code = "R90T92" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "R" and child.code = "R93" CREATE (child)-[:hasParent]->(parent);
+MATCH (parent:`_generic_hierarchy_node_business-investment-activity`), (child:`_generic_hierarchy_node_business-investment-activity`) where parent.code = "R" and child.code = "RTU" CREATE (child)-[:hasParent]->(parent);
```
Comment (eldeal) [inline cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher]: RTU is a child of R? we dont have data for T and U anyway 🤔

**Comment by MikeData** [comment]
this might make this a little easier to follow. These are all the codes used in the larger business investment file. the smaller load file uses a subset of these.

['L', '_T', 'A', 'A01', 'A02', 'A03', 'B', 'BTE', 'C', 'C10T12',
       'C13T15', 'C16', 'C16T18', 'C17', 'C18', 'C19', 'C20', 'C21',
       'C22', 'C22_23', 'C23', 'C24', 'C24_25', 'C25', 'C26', 'C27',
       'C28', 'C29', 'C29_30', 'C30', 'C31T33', 'C31_32', 'C33', 'D', 'E',
       'E36', 'E37T39', 'F', 'G', 'G45', 'G46', 'G47', 'H', 'H49', 'H50',
       'H51', 'H52', 'H53', 'I', 'J', 'J58', 'J58T60', 'J59_60', 'J61',
       'J62_63', 'K', 'K64', 'K65', 'K66', 'M', 'M69T71', 'M69_70', 'M71',
       'M72', 'M73', 'M73T75', 'M74_75', 'M_N', 'N', 'N77', 'N78', 'N79',
       'N80T82', 'O', 'OTQ', 'P', 'Q', 'Q86', 'Q87_88', 'R', 'R90T92',
       'R93', 'RTU', 'S', 'S94', 'S95', 'S96']

File: cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'_T',label:'Total - All activities' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'A',label:'Agriculture, forestry and fishing' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'A01',label:'Crop and animal production, hunting and related service activities' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'A02',label:'Forestry and logging' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'A03',label:'Fishing and aquaculture' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'B',label:'Mining and quarrying' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'BTE',label:'Industry (except construction)' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'C',label:'Manufacturing' });
+CREATE (node:`_generic_hierarchy_node_business-investment-activity` { code:'C10T12',label:'Manufacture of food products; beverages and tobacco products' });
```
[the remark this answers is not in the document]
Comment (MikeData) [reply cmd/hierarchy-transformer/output/businessInvestment/hierarchy.cypher]: no, that'd make far too much sense :)

I've added a list of all codes at the bottom of the PR. Might start doing this in future as standard for hierarchy PRs.

**Pull request merged**